s/one_for_one_test.go

```diff
@@ -0,0 +1,96 @@
+package s_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/capatazlib/go-capataz/s"
+	. "github.com/capatazlib/go-capataz/stest"
+)
+
+func TestSingleFailingChildRecovers(t *testing.T) {
+	parentName := "root"
+	// Fail only one time
+	child1, startFailingChild1 := FailingChild(1, "child1")
+
+	events, err := ObserveSupervisor(
+		context.TODO(),
+		parentName,
+		[]s.Opt{
+			s.WithChildren(child1),
+		},
+		func(em EventManager) {
+			// NOTE: we won't stop the supervisor until the child has failed at least
+			// once
+			evIt := em.Iterator()
+			// 1) Wait till all the three is up
+			evIt.SkipTill(ProcessStarted("root"))
+			// 2) Start the failing behavior of child1
+			startFailingChild1()
+			// 3) Wait till first restart
+			evIt.SkipTill(ProcessStarted("root/child1"))
+		},
+	)
+
+	assert.NoError(t, err)
+
+	AssertExactMatch(t, events,
+		[]EventP{
+			// start children from left to right
+			ProcessStarted("root/child1"),
+			ProcessStarted("root"),
```

Comment:
This is always counter intuitive to me. I would always expect the root to precede the child (prefix). Is this more like infix. prefix or postfix notation?

Reply:
Leafs of the tree are the workers of the system. They start first and once all of them start, the supervisor is considered started (monitoring their errors). The start order will be pre-order (`LeftToRight`) or post-order (`RightToLeft`) depending on the `SupervisorSpec` settings.

We do traverse the root first, but the parents are the last bit that start in the system. If there is an error on any started worker before the supervisor starts, the worker child will block in a write chan and the supervisor will pick it up once it starts.

Reply:
I'm going to write the lines above in the comments so that expectations of the tests becomes very explicit and sense can be built.

Reply:
I see, and you have explained this before. Sorry. What you explain is not wrong, but there may be overloaded concepts, perhaps (???). What is counter intuitive may be this: suppose the the system is up and running (everything marked as started and running smoothly). If a child dies, completely dead (restarts fail, etc), what happens to the root supervisor: marked as in error only, but still running? Is error state equivalent as not running? What if all the other children are still running? Is there a distinction between a running supervisor and its children running? Is there a distinction between the error state of a supervisor and the error state of the children? I am not at all advocating more complexity or any changes. I am just wondering if more clarity in the comments or README would be helpful. I see you have done some of that already and it is great.